### PR TITLE
add an example with next.js and monaco-graphql

### PR DESCRIPTION
**Pull request by acao**
introduces a `monaco-graphql` example using the default `create-next-app` template

### CODE REVIEW

File: examples/monaco-graphql-nextjs/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@graphiql/toolkit": "^0.8.0",
+    "@next/font": "13.1.6",
+    "@types/node": "18.13.0",
+    "@types/react": "18.0.28",
+    "@types/react-dom": "18.0.11",
+    "@zeit/next-css": "^1.0.1",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:17]: this package is deprecated 
image

File: examples/monaco-graphql-nextjs/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@graphiql/toolkit": "^0.8.0",
+    "@next/font": "13.1.6",
+    "@types/node": "18.13.0",
+    "@types/react": "18.0.28",
+    "@types/react-dom": "18.0.11",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:16]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "@zeit/next-css": "^1.0.1",
+    "css-loader": "^6.7.3",
+    "eslint": "8.34.0",
+    "eslint-config-next": "13.1.6",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:20]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    "graphql": "^16.6.0",
+    "graphql-ws": "^5.11.3",
+    "jsonc-parser": "^3.2.0",
+    "marked": "^4.2.12",
+    "mini-css-extract-plugin": "^2.7.2",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:25]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "monaco-editor": "^0.35.0",
+    "monaco-editor-webpack-plugin": "^7.0.1",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:27]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    "monaco-graphql": "^1.1.7",
+    "next": "13.1.6",
+    "next-global-css": "^1.3.1",
+    "prettier": "^2.8.4",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:31]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "style-loader": "^3.3.1",
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:34]: should be in devDeps

File: examples/monaco-graphql-nextjs/package.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    "typescript": "4.9.5",
+    "worker-loader": "^3.0.8"
```
Comment (B2o5T) [inline examples/monaco-graphql-nextjs/package.json:36]: should be in devDeps

**Comment by acao** [comment]
I'm not sure I want to diverge from the next.js template this much. Next doesn't seem to think these packages should be in `devDeps`. This was created from the latest `create-next-app`.

File: examples/monaco-graphql-nextjs/package.json (hunk 1 above)
Comment (acao) [reply examples/monaco-graphql-nextjs/package.json:17]: oops! that was leftover from an experiment, it isn't being used

**Comment by acao** [comment]
this is why I don't think `examples/` should be linted or should follow a common standard with the rest of the application. they are often generated from other tools, and follow their own standards, which is just more maintenance surface area.  if they weren't so useful for cross-workspace project references watch reference development and for bundler validation, I would suggest moving them to a seperate standalone repository

**Comment by acao** [comment]
this is also why this example has a package-lock.json, it seems the `create-next-app` defaults to `npm`

File: examples/monaco-graphql-nextjs/package.json (hunk 6 above)
Comment (acao) [reply examples/monaco-graphql-nextjs/package.json:27]: prettier is a peer dependency of `monaco-graphql`

**Pull request merged**